### PR TITLE
Start the initial draft implementation

### PR DESCRIPTION
**Pull request by edersonmf**
This includes options to manage configuration as well as do simple
queries.

### CODE REVIEW

**Review by ddmendes — APPROVED** [review]

**Review by robsondepaula — APPROVED** [review]
LGTM!

File: pkg/config/file.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  jsonFile, err := os.Open(homeDir() + configFileName)
+  if err != nil && os.IsNotExist(err) {
+    return ConfigurationFile{
+      Servers: map[string]Server{},
+      Roles: map[string]Role{},
+    }
+  } else {
+    checkError(err)
+  }
+  byteValue, _ := ioutil.ReadAll(jsonFile)
```
Comment (ddmendes) [inline pkg/config/file.go]: The `defer jsonFile.Close()` should be the first thing after checking file errors. So past error checking you always guarantee the file will be closed.

File: pkg/config/file.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  var configFile ConfigurationFile
+  err = json.Unmarshal(byteValue, &configFile)
```
Comment (ddmendes) [inline pkg/config/file.go]: This is not wrong but there's another approach for Unmarshaling. Once os.File has the Read() method then it implements io.Reader. Thus you can use
```suggestion
  err = json.NewDecoder(jsonFile).Decode(&configFile)
```
and save the file reading segment. But it's up to you. Both approaches are right.

File: pkg/options/configure_option.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+func addServer(configFile *config.ConfigurationFile) {
+  reader := bufio.NewReader(os.Stdin)
+  fmt.Print("Server name: ")
+  serverName, _ := reader.ReadString(lineBreakAsByte)
+  serverName = strings.TrimSuffix(serverName, lineBreak)
+  configFile.Servers[serverName] = buildServer(reader)
+  fmt.Print("Set as default? [Y/n]: ")
+  defaultServer, _ := reader.ReadString(lineBreakAsByte)
+  defaultServer = strings.TrimSuffix(defaultServer, lineBreak)
+  if len(configFile.CurrentServer) <=  0 || len(defaultServer) <= 0 || defaultServer == "Y" {
```
Comment (ddmendes) [inline pkg/options/configure_option.go]: You could also expect for `"y"`

File: pkg/options/configure_option.go [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+func addRole(configFile *config.ConfigurationFile)  {
+  reader := bufio.NewReader(os.Stdin)
+  fmt.Print("Role name: ")
+  roleName, _ := reader.ReadString(lineBreakAsByte)
+  roleName = strings.TrimSuffix(roleName, lineBreak)
+  configFile.Roles[roleName] = buildRole(reader)
+  fmt.Print("Set as default? [Y/n]: ")
+  defaultRole, _ := reader.ReadString(lineBreakAsByte)
+  defaultRole = strings.TrimSuffix(defaultRole, lineBreak)
+  if len(configFile.CurrentRole) <=  0 || len(defaultRole) <= 0 || defaultRole == "Y" {
```
Comment (ddmendes) [inline pkg/options/configure_option.go]: You could also expect for `"y"`

File: pkg/options/use_option.go [hunk 5]
```diff
@@ -0,0 +1,19 @@
+package options
+
+func (u *UseCmd) Run(ctx *Context) error {
```
Comment (ddmendes) [inline pkg/options/use_option.go:7]: We could validate if Server/Role exists in the configuration before setting it as default

File: pkg/options/search_option.go [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+    out, err := buildFromTemplate("query", queryClauseTemplate, s)
+    if err != nil {
+      return err
+    }
+    clause = out.String()
+  }
+
+  server := ctx.ConfigFile.Servers[ctx.ConfigFile.CurrentServer]
+  if len(s.Server) > 0 {
+   server = ctx.ConfigFile.Servers[s.Server]
```
Comment (ddmendes) [inline pkg/options/search_option.go]: You could easily check Server existence by
```suggestion
   server, ok = ctx.ConfigFile.Servers[s.Server]
   if !ok {
       return fmt.Errorf("Server %s does not exist", s.Server)
   }
```

**Review by ddmendes — CHANGES_REQUESTED** [review]
Very good. There is just some minor details.

**Pull request merged**